--- pkg-config-1.4.0/lib/pkg-config.rb.orig	2020-02-05 17:56:08.247644168 +0100
+++ pkg-config-1.4.0/lib/pkg-config.rb	2020-02-06 16:16:50.538517573 +0100
@@ -249,13 +249,15 @@
     @path_position
   end
 
-  def collect_requires(&block)
+  def collect_requires(seen=[], &block)
     packages = []
     targets = yield(self)
     targets.each do |name|
-      package = self.class.new(name, @options)
-      packages << package
-      packages.concat(package.collect_requires(&block))
+      if !seen.include?(name)
+        package = self.class.new(name, @options)
+        packages << package
+        packages.concat(package.collect_requires(seen + [name], &block))
+      end
     end
     packages_without_self = packages.reject do |package|
       package.name == @name
