--- pkg-config-1.1.5/lib/pkg-config.rb~	2015-12-19 22:10:31.000000000 +0100
+++ pkg-config-1.1.5/lib/pkg-config.rb	2015-12-19 22:16:52.177522169 +0100
@@ -222,7 +222,7 @@
     all_libs = all_libs.join(" ").gsub(/-([Ll]) /, '\1').split.uniq
     path_flags, other_flags = all_libs.partition {|flag| /\A-L/ =~ flag}
     path_flags = path_flags.reject do |flag|
-      /\A-L\/usr\/lib(?:64)?\z/ =~ flag
+      /\A-L\/usr\/lib(?:64|x32)?\z/ =~ flag
     end
     if @msvc_syntax
       path_flags = path_flags.collect do |flag|
@@ -283,11 +283,13 @@
   def guess_default_path
     arch_depended_path = Dir.glob('/usr/lib/*/pkgconfig').join(SEPARATOR)
     default_path = ["/usr/local/lib64/pkgconfig",
+		    "/usr/local/libx32/pkgconfig",
                     "/usr/local/lib/pkgconfig",
                     "/usr/local/libdata/pkgconfig",
                     "/opt/local/lib/pkgconfig",
                     arch_depended_path,
                     "/usr/lib64/pkgconfig",
+                    "/usr/libx32/pkgconfig",
                     "/usr/lib/pkgconfig",
                     "/usr/X11/lib/pkgconfig/",
                     "/opt/X11/lib/pkgconfig/",
@@ -302,6 +304,7 @@
       Dir.glob((pkg_config_prefix + "lib/*/pkgconfig").to_s).join(SEPARATOR)
     [pkg_config_arch_depended_path,
      (pkg_config_prefix + "lib64/pkgconfig").to_s,
+     (pkg_config_prefix + "libx32/pkgconfig").to_s,
      (pkg_config_prefix + "lib/pkgconfig").to_s,
      (pkg_config_prefix + "libdata/pkgconfig").to_s,
      default_path].join(SEPARATOR)
